itou/users/models.py

```diff
@@ -1073,6 +1082,39 @@ def clean_pole_emploi_fields(cleaned_data):
             # the object is shared between the caller and the called routine.
             cleaned_data["lack_of_pole_emploi_id_reason"] = ""
 
+    @staticmethod
+    def clean_nir_title_birthdate_fields(cleaned_data, remind_nir_in_error=False):
+        """
+        Validate consistency between NIR, title and birthdate
+        """
+        if cleaned_nir := cleaned_data.get("nir"):
+            if cleaned_title := cleaned_data.get("title"):
+                if (cleaned_nir[0] == "1" and cleaned_title != Title.M) or (
+                    cleaned_nir[0] == "2" and cleaned_title != Title.MME
+                ):
+                    raise ValidationError(
+                        JobSeekerProfile.ERROR_JOBSEEKER_INCONSISTENT_NIR_TITLE
+                        + (
+                            f" Le numéro de sécurité social enregistré est {cleaned_nir}."
+                            if remind_nir_in_error
+                            else ""
+                        )
+                    )
+            if cleaned_birthdate := cleaned_data.get("birthdate"):
+                nir_year = cleaned_nir[1:3]
+                nir_month = cleaned_nir[3:5]
+                birthdate_year = cleaned_birthdate.strftime("%y")
+                birthdate_month = cleaned_birthdate.strftime("%m")
+                if nir_year != birthdate_year or nir_month != birthdate_month:
```

Comment:
```suggestion
                nir_year_month = cleaned_nir[1:5]
                birthdate_year_month = cleaned_birthdate.strftime("%y%m")
                if nir_year_month != birthdate_year_month:
```
si on ne fait pas de message d'erreur différent pour des erreurs sur le mois/l'année/les deux.